### ISSUE TITLE
The stream I get from iosrtc and pass to peerJs gives error

### ISSUE THREAD

**Issue by uhotspot4** [peers/peerjs#705]
I have my voice chat working perfectly using PeerJs but since WKWebView on iOS doesn't provide navigator.getUserMedia, had to use something like cordova-plugin-iosrtc just to get the userMedia, but still I use PeerJs for the rest of the process. And now I get following error:

`TypeError: Argument 1 ('track') to RTCPeerConnection.addTrack must be an instance of MediaStreamTrack`

And following is the piece of code I am using:

     cordova.plugins.iosrtc.getUserMedia({ audio: true, video: false },
        ).then(function (stream) {
        try{
           //Folowing is the line causing error
          var call = peer.call('A', stream);

          }catch(error) {console.log("This is the error I get:"+error)}

        call.on('stream', function(remoteStream) {  
           document.getElementById('audio').srcObject = remoteStream;
        });
    }).catch(function (error) {
      console.error('getUserMedia failed: ', error);
    }

I would appreciate if someone tells me how I can combine those two libraries and what I have done wrong!

**Comment by mozhn** [comment]
Would you try as below:

`navigator.mediaDevices.getUserMedia({
      audio: true, video: true
    })
`

**Comment by uhotspot4** [comment]
@mozhn Thanks for the reply. I did and it gives the error saying mediaDevices is undefined. 

I found same error being reported here [https://github.com/webrtcHacks/adapter/issues/989](url) and the fixes @ [https://github.com/webrtcHacks/adapter/pull/990/files/588953a348f7be9088f49899cb028a5a89316ec1](url) and [https://github.com/webrtcHacks/adapter/pull/992/files/f18c41e39c2c65186954341b0581e8a0e31c9033](url).

I printed out the window.RTCPeerConnection.prototype.addTrack right before calling peer.call('A', stream); and it looks exactly as what they have in that patch!

    function addTrack(track) {
       var stream = arguments[1];

      if (stream) {
        if (!this._localStreams) {
          this._localStreams = [stream];
        } else if (!this._localStreams.includes(stream)) {
          this._localStreams.push(stream);
        }
      }

      return _addTrack.apply(this, arguments);
    }

**Comment by mozhn** [comment]
Could there be something missing. My project Ionic 4 and Cordova. I'm not getting error. I'm not sure but did you import the library below?

https://webrtc.github.io/adapter/adapter-latest.js

My index.html source:
https://hastebin.com/nadiqarave.xml

I hope it helps.

**Comment by uhotspot4** [comment]
@mozhn Didn't even know about adapter.js. Just came across it when searching for the error I get on Internet. So do you think I can drop iosrtc and instead user adapter.js? 

Since you are using cordova I assume yours is an app using WKWebView too and not just a webpage running in safari, am I right? 
Didn't you ran into this problem when accessing mediaStream? Apparently it is a bug in webkit implementation in IOS and they have reported a bug for it and it hasn't been resolved yet -> [https://bugs.webkit.org/show_bug.cgi?id=208667](url)

**Comment by mozhn** [comment]
Yes i used WKWebView. I have never experienced the error you mentioned. I followed the steps at this address and succeeded. The bottom of the page.

https://github.com/cordova-rtc/cordova-plugin-iosrtc

Example:
document.addEventListener('deviceready', function () {
  // Just for iOS devices.
  if (window.device.platform === 'iOS') {
    cordova.plugins.iosrtc.registerGlobals();

    // load adapter.js
    var adapterVersion = 'latest';
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://webrtc.github.io/adapter/adapter-" + adapterVersion + ".js";
    script.async = false;
    document.getElementsByTagName("head")[0].appendChild(script);
  }
});`

Now I realized it was just for cordova.
`

**Comment by uhotspot4** [comment]
@mozhn Thanks a lot, I'll check it out.

**Comment by mozhn** [comment]
Yes it is

**Comment by uhotspot4** [comment]
It is working now. Thanks for all the help.

**Comment by mozhn** [comment]
It will be useful for everyone if you explain exactly what you are doing.

**Comment by dyegonery** [comment]
@uhotspot4 I had the same problem, when trying to use this plugin with Peerjs on an Ionic app. Any insight on how you solved this problem?

**Comment by hthetiot** [comment]
See this comment: https://github.com/peers/peerjs/issues/705#issuecomment-660676552

I recommand to close this issue, for future issue please create issue on cordova-plugin-iosrtc repository itself https://github.com/cordova-rtc/cordova-plugin-iosrtc

Notice that I'm the maintainer of iosRTC.
